src/App.tsx

```diff
@@ -262,6 +272,50 @@ function AppContent() {
     }
   }, [type]);
 
+  useEffect(() => {
+    if (showUnitSnackbar) {
+      snackbar.pageInfo(
+        strings.formatString<any>(
+          strings.UNITS_INITIALIZED_MESSAGE,
+          userPreferences.preferredWeightSystem ?? 'metric',
+          <Link to={APP_PATHS.MY_ACCOUNT}>{strings.MY_ACCOUNT}</Link>
+        ),
+        strings.UNITS_INITIALIZED_TITLE,
+
+        {
+          label: strings.GOT_IT,
+          apply: () => {
+            updatePreferences('unitsAcknowledgedOnMs', Date.now());
+          },
+        },
+        'user'
+      );
+    }
+  }, [showUnitSnackbar, snackbar, userPreferences.preferredWeightSystem]);
+
+  useEffect(() => {
+    const initializaUnit = async () => {
+      if (!user || !userPreferences || !pathname) {
+        return;
+      }
+
+      if (!userPreferences.preferredWeightSystem) {
```

Comment:
seems like we need to update the userPreferences in the provider after the update on line 303, or we may repeatedly set the preference on every useEffect? may need to confirm